sympy/core/relational.py

```diff
@@ -1138,3 +1033,308 @@ def _eval_relation(cls, lhs, rhs):
     '<': StrictLessThan,
     'lt': StrictLessThan,
 }
+
+
+def _n2(a, b):
+    """Return (a - b).evalf(2) if a and b are comparable, else None.
+    This should only be used when a and b are already sympified.
+    """
+    # /!\ it is very important (see issue 8245) not to
+    # use a re-evaluated number in the calculation of dif
+    if a.is_comparable and b.is_comparable:
+        dif = (a - b).evalf(2)
+        if dif.is_comparable:
+            return dif
+
+
+def fuzzy_resolve(lhs, rhs, op, cls):
+    """
+    Resolves op to S.true if cmp returns true.
+    or to S.false if cmp returns false
+    or to an unevaluated if cmp returns none.
+    """
+    val = op(lhs, rhs)
+    if val is None:
+        return cls(lhs, rhs, evaluate=False)
+    elif isinstance(val,bool):
+        if val is True:
+            return S.true
+        else:
+            return S.false
+    else:
+        return val
```

Comment:
Should this case not be an error? Can op return anything other than True, False or None?